scripts/basic-workflow-tests.sh

```diff
@@ -119,51 +114,96 @@ test_init() {
+    elif [ "${MIXTRAL}" -ne 1 ] && [ "${GRANITE}" -ne 1 ] && [ "$BACKEND" == "vllm" ]; then
+        sed -i -e 's|merlinite.*|instructlab/merlinite-7b-lab|' "${CONFIG_HOME}/instructlab/config.yaml"
     fi
 }
 
 test_download() {
     task Download the model
 
     if [ "$GRANITE" -eq 1 ]; then
-        step Downloading the granite model
-        ilab model download --repository instructlab/granite-7b-lab-GGUF --filename granite-7b-lab-Q4_K_M.gguf
-    elif [ "$BACKEND" = "vllm" ]; then
-        step Downloading the model for vLLM
-        ilab download --repository instructlab/merlinite-7b-lab
+        if [ "$BACKEND" == "vllm" ]; then
+            step Downloading the .safetensors granite model
+            ilab download --repository instructlab/granite-7b-lab
+        else
+            step Downloading the granite .gguf model
+            ilab model download --repository instructlab/granite-7b-lab-GGUF --filename granite-7b-lab-Q4_K_M.gguf
+        fi
     elif [ "$MIXTRAL" -eq 1 ]; then
-        step Downloading the mixtral model
-        ilab model download --repository TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF --filename mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf --hf-token "${HF_TOKEN}"
+        if [ "$BACKEND" == "vllm" ]; then
+            step Downloading the .safetensors mixtral model
+            ilab model download --repository TheBloke/Mixtral-8x7B-Instruct-v0.1-GPTQ
+        else
+            step Downloading the mixtral .gguf model
+            ilab model download --repository TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF --filename mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf --hf-token "${HF_TOKEN}"
+        fi
     else
-        step Downloading the default model
-        ilab model download
+        if [ "$BACKEND" == "vllm" ]; then
+            step Downloading the default .safetensors model to use with vLLM
+            ilab download --repository instructlab/merlinite-7b-lab
+        else
+            step Downloading the default model
+            ilab model download
+        fi
     fi
 
     if [ "$EVAL" -eq 1 ]; then
-        step Downloading a model to use with evaluation
-        ilab model download --repository instructlab/granite-7b-lab
+        if [ "$GRANITE" -ne 1 ]; then
+            step Downloading a granite model to use with evaluation
+            ilab model download --repository instructlab/granite-7b-lab
+        fi
+
+        # if vllm is not configured as the backend, merlinite-7b-lab
+        # needs to be downloaded in .safetensor format since evaluation
+        # can only evaluate .safetensors models and merlinite-7b-lab is needed
+        if [ "$BACKEND" == "llama-cpp" ]; then
+            step Downloading a merlinite-7b-lab to use with evaluation
+            ilab download --repository instructlab/merlinite-7b-lab
+        fi
     fi
 }
 
 test_serve() {
+    # TODO: are these assignments even necessary because model_path is already set in the config in test_init
```

Comment:
(To answer the question) looks like this function is used with and without `$1`. It is probably redundant to pass `--model-path` when `$1` is not passed (then the values from the config file will do), otherwise `--model-path` should be added.

So: probably lines 204-206 should be changed so that they kick in only when `$1` is passed; otherwise omit the option.